entity/src/users.rs

```diff
@@ -2,26 +2,27 @@
 
 use crate::Id;
 use axum_login::AuthUser;
-use chrono::{DateTime, Utc};
 use sea_orm::entity::prelude::*;
 use serde::{Deserialize, Serialize};
 
-#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Eq, Deserialize, Serialize)]
+#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Eq, Serialize, Deserialize)]
 #[sea_orm(schema_name = "refactor_platform", table_name = "users")]
 pub struct Model {
     #[sea_orm(primary_key)]
     #[serde(skip_deserializing)]
     pub id: Id,
-    #[sea_orm(unique, indexed)]
+    #[sea_orm(unique)]
+    pub external_id: Uuid,
+    #[sea_orm(unique)]
     pub email: String,
-    pub first_name: String,
-    pub last_name: String,
-    pub display_name: String,
+    pub first_name: Option<String>,
+    pub last_name: Option<String>,
+    pub display_name: Option<String>,
     pub password: String,
-    pub github_username: String,
-    pub github_profile_url: String,
-    pub created_at: DateTime<Utc>,
-    pub updated_at: DateTime<Utc>,
+    pub github_username: Option<String>,
+    pub github_profile_url: Option<String>,
+    pub created_at: Option<DateTimeWithTimeZone>,
+    pub updated_at: Option<DateTimeWithTimeZone>,
```

Comment:
Can we make these two timestamps non-optional? I don't think it makes sense to ever allow for a `Users` entity to be created without these being set, especially since the DB can even set the default to `now()` if our Rust code doesn't explicitly set it on creation.

Reply:
Updated in 360ce0c107cb7c45196f1823a05dfde432480c54!